Bugfix/zenko 4813

williamlardier opened the pull request:
WIP

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4813 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.8.33`

* `2.9.6`


Please check the `Fix Version/s` of ZENKO-4813, or the target
branch of this pull request.

Seragonia committed:
ZENKO-4813: bump CTST

Seragonia committed:
ZENKO-4813: move logger to Zenko class

Seragonia committed:
ZENKO-4813: let extractPropertyFromResults throw not to hide errors

Seragonia committed:
ZENKO-4813: move unused util functions to the util file

Seragonia committed:
ZENKO-4813: consistent world variable name

Seragonia committed:
ZENKO-4813: typed generic S3 function calls

Seragonia committed:
ZENKO-4813: use CTST standard Command type, remove ApiResult

Seragonia committed:
ZENKO-4813: remove 'options' from Zenko class

Seragonia committed:
ZENKO-4813: use type from CTST

Seragonia committed:
ZENKO-4813: misc improvements to reduce build times

Seragonia committed:
ZENKO-4813: use standard Role type

Seragonia committed:
ZENKO-4813: remove legacy identity logic

CTST used to embed the current identity, without support for
multiple ones at once, in the Cache (both as a cache key, or in
the cloned World's parameters). Eventually we end up with code
where the identity management is not clear: we have the cliMode,
the world.parameters, Cache.parameters and Cache.AccessKeys.
In the new version, CTST introduces a new class, Identity, that
manages all the identities of different types, and used in every
API calls dynamically. This class serves as a store and cache of
all identities created within a cucumberJS process, but also as
a way to switch between identities freely and more simply, while
abstracting the CLI logic  (the legacy "cliMode" option).

Seragonia committed:
ZENKO-4813: init logger in the world class

Seragonia committed:
ZENKO-4813: accept that extract property returns null objects

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4813 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.8.33`

* `2.9.7`


Please check the `Fix Version/s` of ZENKO-4813, or the target
branch of this pull request.

Seragonia committed:
ZENKO-4813: rework identities in steps implementation

The old implementation was using the old flow, hence it implements
the identity mangement in a non-consistent way. This commit
is a clean up and fix of the current implementation.

Seragonia committed:
ZENKO-4813: remove duplicate timeout logic

Seragonia committed:
ZENKO-4813: use standard types

Seragonia committed:
ZENKO-4813: move quota hook in hook file

Seragonia committed:
ZENKO-4813: cleanup Zenko class

Seragonia committed:
ZENKO-4813: add a missing permission to the delete quota scenario

Seragonia committed:
ZENKO-4813: consume all roles when calling GRFWI

Seragonia committed:
ZENKO-4813: fix tests with missing permissions or steps
